### PR TITLE
chore: upgrade lightning and lightning-invoice deps

### PR DESCRIPTION
**Pull request by tvolk131**
Part of #4561 and #2356

### CODE REVIEW

File: fedimint-core/src/bitcoin_migration.rs [hunk 1]
```diff
@@ -155,6 +183,12 @@ pub fn bitcoin29_to_bitcoin30_sha256_hash(
     bitcoin30::hashes::sha256::Hash::from_byte_array(hash.into_inner())
 }
 
+pub fn bitcoin30_to_bitcoin29_sha256_hash(
+    hash: bitcoin30::hashes::sha256::Hash,
+) -> bitcoin::hashes::sha256::Hash {
+    bitcoin::hashes::sha256::Hash::from_str(&hash.to_string()).unwrap()
```
Comment (elsirion) [inline fedimint-core/src/bitcoin_migration.rs:189]: Why always strings and not bytes :sob: there probably is some `into_inner`/`to_bytes` function.

File: fedimint-testing/src/ln/mock.rs [hunk 2]
```diff
@@ -67,19 +70,26 @@ impl LightningTest for FakeLightningTest {
...
     ) -> ln_gateway::Result<Bolt11Invoice> {
-        let ctx = bitcoin::secp256k1::Secp256k1::new();
+        let ctx = bitcoin30::secp256k1::Secp256k1::new();
 
         Ok(InvoiceBuilder::new(Currency::Regtest)
             .description("".to_string())
-            .payment_hash(sha256::Hash::hash(&[0; 32]))
+            .payment_hash(bitcoin29_to_bitcoin30_sha256_hash(sha256::Hash::hash(
+                &[0; 32],
+            )))
```
Comment (elsirion) [inline fedimint-testing/src/ln/mock.rs:79]: You can just construct the `bitcoin30` hash right away here afaik?

Comment (tvolk131) [reply fedimint-testing/src/ln/mock.rs]: Done.

File: fedimint-core/src/bitcoin_migration.rs (hunk 1 above)
Comment (tvolk131) [reply fedimint-core/src/bitcoin_migration.rs]: Ope 🙊 my bad! Fixed.

**Comment by tvolk131** [comment]
Closing now that we merged https://github.com/fedimint/fedimint/pull/5118 which also does this

**Pull request closed without merging**